src/components/delete-popup.tsx

```diff
@@ -0,0 +1,51 @@
+'use client'
+
+import { Trash2 } from 'lucide-react'
+import { AsyncButton } from './async-button'
+import { Button } from './ui/button'
+import {
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogDescription,
+  DialogOverlay,
+  DialogPortal,
+  DialogTitle,
+  DialogTrigger,
+} from './ui/dialog'
+
+export function DeletePopup({ onDelete }: { onDelete: () => Promise<void> }) {
+  return (
+    <Dialog>
+      <DialogTrigger asChild>
+        <Button variant="destructive">
+          <Trash2 className="w-4 h-4 mr-2" />
+          Delete
+        </Button>
+      </DialogTrigger>
+      <DialogPortal>
+        <DialogOverlay />
+        <DialogContent>
```

Comment:
You probably should remove the enclosing `<DialogPortal>` as well as the following `<DialogOverlay />`, since both elements are already included in `<DialogContent />`.

So basically you are adding them twice. Just having one `<DialogOverlay />` also makes the background a little bit transparent which looks better, too.